File: MaxConsecutiveOnesIII.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int left = 0, right = 0;
        int maxLength = 0;
        int length = 0;

        for(;right < nums.size();right++){
            if(nums[right])
                length++;
            
            if((right - left + 1) - length > k){
                if(nums[left++])
                    length--;
            }
            else{
                maxLength = max(maxLength, (right - left + 1));
            }
        }
        
        return maxLength;
    }
};

File: phase_one_eduacation/week4/leetcode/max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        left = 0
        longest = 0
        length = 0
        for right in range(len(nums)):
            if nums[right] == 1:
                length += 1
            
            currWindow = right - left + 1
            if currWindow - length > k:
                if nums[left] == 1:
                    length -= 1
                left += 1
            else:
                longest = max(longest, currWindow)
            
        return longest